src/EnergyPlus/VentilatedSlab.cc

```diff
@@ -1539,19 +1538,19 @@ namespace VentilatedSlab {
             state.dataVentilatedSlab->MySizeFlag.allocate(state.dataVentilatedSlab->NumOfVentSlabs);
             state.dataVentilatedSlab->MyPlantScanFlag.allocate(state.dataVentilatedSlab->NumOfVentSlabs);
             state.dataVentilatedSlab->MyZoneEqFlag.allocate(state.dataVentilatedSlab->NumOfVentSlabs);
-            state.dataVentilatedSlab->ZeroSourceSumHATsurf.dimension(state.dataGlobal->NumOfZones, 0.0);
-            state.dataVentilatedSlab->QRadSysSrcAvg.dimension(state.dataSurface->TotSurfaces, 0.0);
-            state.dataVentilatedSlab->LastQRadSysSrc.dimension(state.dataSurface->TotSurfaces, 0.0);
-            state.dataVentilatedSlab->LastSysTimeElapsed.dimension(state.dataSurface->TotSurfaces, 0.0);
-            state.dataVentilatedSlab->LastTimeStepSys.dimension(state.dataSurface->TotSurfaces, 0.0);
 
-            // Initialize total areas for all radiant systems
+            // Initialize total areas for all radiant systems and dimension record keeping arrays
             for (RadNum = 1; RadNum <= state.dataVentilatedSlab->NumOfVentSlabs; ++RadNum) {
                 state.dataVentilatedSlab->VentSlab(RadNum).TotalSurfaceArea = 0.0;
-                for (SurfNum = 1; SurfNum <= state.dataVentilatedSlab->VentSlab(RadNum).NumOfSurfaces; ++SurfNum) {
-                    state.dataVentilatedSlab->VentSlab(RadNum).TotalSurfaceArea +=
-                        state.dataSurface->Surface(state.dataVentilatedSlab->VentSlab(RadNum).SurfacePtr(SurfNum)).Area;
+                auto &numRadSurfs = state.dataVentilatedSlab->VentSlab(RadNum).NumOfSurfaces;
+                auto &thisVentSlab = state.dataVentilatedSlab->VentSlab(RadNum);
+                for (SurfNum = 1; SurfNum <= numRadSurfs; ++SurfNum) {
+                    thisVentSlab.TotalSurfaceArea += state.dataSurface->Surface(thisVentSlab.SurfacePtr(SurfNum)).Area;
                 }
+                thisVentSlab.QRadSysSrcAvg.dimension(numRadSurfs, 0.0);
+                thisVentSlab.LastQRadSysSrc.dimension(numRadSurfs, 0.0);
+                thisVentSlab.LastSysTimeElapsed = 0.0;
+                thisVentSlab.LastTimeStepSys = 0.0;
             }
```

Comment:
not sure

Reply:
@rraustad I was confused by the numbers...what were they supposed to indicate?  In any case, I went through the stuff that you fixed...all pretty solid stuff.  Shocking that one section of code just got axed and apparently was never needed?  Thanks for your thorough review and getting in the trenches on this one.

Reply:
@RKStrand I ran local CppCheck on this code and it pointed me to variables that were initialized and never used. That's how that chunk of code got axed.